internal/components/installation_backup.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package components
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/mattermost/mattermost-cloud/internal/webhook"
+	"github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type installationBackupStore interface {
+	IsInstallationBackupRunning(installationID string) (bool, error)
+	CreateInstallationBackup(backup *model.InstallationBackup) error
+	GetWebhooks(filter *model.WebhookFilter) ([]*model.Webhook, error)
+}
+
+// TriggerInstallationBackup verifies that backup can be started for an Installation and triggers it.
+func TriggerInstallationBackup(store installationBackupStore, installation *model.Installation, env string, logger log.FieldLogger) (*model.InstallationBackup, error) {
+	if err := model.EnsureInstallationReadyForBackup(installation); err != nil {
```

Comment:
Slight styling nit in that we don't normally condense the err check to one line in `mattermost-cloud`, but if we want to change this guideline I am okay with it. Just want it to be a conscious choice by the team if we do.

Reply:
Oh, 0/5 on that, I probably just copied it from some PoC I was doing prior. No need to change convention imo.